node/src/components/small_network/metrics.rs

```diff
@@ -212,6 +255,17 @@ impl Metrics {
             debug!("not recording metrics, component already shut down");
         }
     }
+
+    /// Creates a set of outgoing metrics that is connected to this set of metrics.
+    pub(super) fn create_outgoing_metrics(&self) -> OutgoingMetrics {
```

Comment:
So we now have the same metrics in two places? Can it happen that the same event will update them twice b/c it will be handled on multiple layers that each can modify it?

Reply:
Sorry, GH was down and unknowningly I've created 3 copies of the same comment.